cmd/renterd/main.go

```diff
@@ -523,7 +529,6 @@ func main() {
 	setupWorkerFn := node.NoopFn
 	if len(cfg.Worker.Remotes) == 0 {
 		if cfg.Worker.Enabled {
-			workerAddr := cfg.HTTP.Address + "/api/worker"
```

Comment:
To avoid too many `/api/worker` strings floating around I'd keep this line here.

Then below do

```go
var workerAddrExtern string // slight rename to indicate this is the worker's external address
if cfg.Bus.RemoveAddr != "" {
  workerAddrExtern = cfg.Worker.ServiceAddress + "/api/worker"
} else {
  workerAddrExtern = workerAddr
}
...

return setupFn(ctx, workerAddrExtern, cfg.HTTP.Password)
...

wc := worker.NewClient(workerAddr, cfg.HTTP.Password)
```

I'd also consider changing `ServiceAddress` to `ExternalAddress` or something similar to better indicate the purpose of the setting.